clients/src/test/java/org/apache/kafka/clients/ClientUtilsTest.java

```diff
@@ -29,12 +29,18 @@ public void testParseAndValidateAddresses() {
         check("mydomain.com:8080");
         check("[::1]:8000");
         check("[2001:db8:85a3:8d3:1319:8a2e:370:7348]:1234", "mydomain.com:10000");
+        check("some.invalid.hostname.foo.bar:9999", "mydomain.com:10000");
```

Comment:
Given that this is now supposed to return a specific subset, you might want to update `check()` to return the list so the tests can validate the output.